test/connectors/test_torch_connector.py

```diff
@@ -659,6 +659,205 @@ def test_circuit_qnn_batch_gradients(self, config):
+            ) from ex
+
+        num_inputs = 2
+
+        # set up QNN
+        qnn = TwoLayerQNN(
+            num_qubits=num_inputs,
+            quantum_instance=self.sv_quantum_instance,
+            input_gradients=True,  # for hybrid qnn
+        )
+
+        # set up dummy hybrid PyTorch module
+        class Net(Module):
+            """Pytorch nn module."""
+
+            def __init__(self):
+                super().__init__()
+                self.fc1 = Linear(4, 2)
+                self.qnn = TorchConnector(
+                    qnn, np.array([0.1] * qnn.num_weights)
+                )  # Apply torch connector
+                self.fc2 = Linear(1, 1)  # 1-dimensional output from QNN
+
+            def forward(self, x):
+                """Forward pass."""
+                x = F.relu(self.fc1(x))
+                x = self.qnn(x)  # apply QNN
+                x = self.fc2(x)
+                return cat((x, 1 - x), -1)
+
+        model = Net()
+
+        # random data set
+        x = Tensor(np.random.rand(5, 4))
+        y = Tensor(np.random.rand(5, 2))
+
+        # define optimizer and loss
+        optimizer = SGD(model.parameters(), lr=0.1)
+        f_loss = MSELoss(reduction="sum")
+
+        # loss and gradients without batch
+        optimizer.zero_grad(set_to_none=True)
+        sum_of_individual_losses = 0.0
+        for x_i, y_i in zip(x, y):
+            output = model(x_i)
+            sum_of_individual_losses += f_loss(output, y_i)
+        sum_of_individual_losses.backward()
+        sum_of_individual_gradients = 0.0
+        for n, param in model.named_parameters():
+            name = n.split(".")
+            # make sure gradient is not None
+            self.assertFalse(param.grad is None)
+            if name[1] == "weight" or name[1] == "_weights":
+                sum_of_individual_gradients += np.sum(param.grad.numpy())
+
+        # loss and gradients with batch
+        optimizer.zero_grad(set_to_none=True)
+        output = model(x)
+        batch_loss = f_loss(output, y)
+        batch_loss.backward()
+        batch_gradients = 0.0
+        for n, param in model.named_parameters():
+            name = n.split(".")
+            # make sure gradient is not None
+            self.assertFalse(param.grad is None)
+            if name[1] == "weight" or name[1] == "_weights":
+                batch_gradients += np.sum(param.grad.numpy())
+
+        # making sure they are equivalent
+        self.assertAlmostEqual(
+            np.linalg.norm(sum_of_individual_gradients - batch_gradients), 0.0, places=4
+        )
+
+        self.assertAlmostEqual(
+            sum_of_individual_losses.detach().numpy(),
+            batch_loss.detach().numpy(),
+            places=4,
+        )
+
+    @data(
+        # output_shape, interpret
+        (2, lambda x: "{:b}".format(x).count("1") % 2),
+    )
+    @requires_extra_library
+    def test_circuit_qnn_hybrid_batch_gradients(self, config):
```

Comment:
Since there's just one tuple in `@data`, parameters can be safely moved in the test itself.

Reply:
True, I will remove that line.